### ISSUE TITLE
Use Cases

### ISSUE THREAD

**Issue by ericgio** [yekrojam/api#54]
I want to make sure we're talking about the same things when discussing use cases. Feel free to clarify, disagree, add to these, etc. Here are the main cases as I think we've been discussing them:

1) **Multi-tenant SaaS**: Standardized Client/UI built on top of the API. Users can switch between orgs.
2) **Single-tenant SaaS**: Self-hosted (ie: custom domain), custom client, built on central API/DB. Users are always in the context of that org.
3) **Standalone**: This is the case where someone just clones the client and API code, spins up their own DB, and has total control over everything. I don't really see this as a long-term solution, just a short-term path to getting someone using the client.

CMS services seem to have a pretty analogous model to how we're thinking about things. If you [look at Wordpress](https://wordpress.com/#plans) for example, they have a free plan that's hosted at wordpress.com, while the paid plans offer custom domains. For the paid plans, you can either use frontend templates or use their API as a "headless CMS" with a custom frontend. Main difference with our model is the concept of linking all the orgs a user is part of.

**Questions**
- A user is a member of two orgs. Org1 uses first model (multi-tenant) while Org2 uses model 2 (single-tenant). If the user logs into the MT site, would they be aware of Org2, and if so, in what ways?

**Comment by ericgio** [comment]
Man, the more I think about this stuff, the more I get confused... For case 1, I've been assuming there is a shared DB across all the orgs and users, right? I was thinking that case 2 would tap into that same dataset, but just be scoped to a single org. It could also just be a SaaS version of case 3, where there'd be a common API configured to an org-specific DB instance. In case 2, the hosting piece is sort of irrelevant; a customer would theoretically just pay for API access and build whatever/wherever they want on top of it.

**Comment by makinde** [comment]
Yeah, I can find some time to leave some thoughts. In some sense, I have a relatively simple model for this in my head. That said, the more (1) diverges from the rest of the list, the harder the current approach becomes. This is effectively the same discussion we're having when talking about user permissions.
